tests/basecli_test.go

```diff
@@ -0,0 +1,222 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	//"strings"
+	"testing"
+	"time"
+)
+
+// Tests assume the `basecoind` and `basecli` binaries
+// have been built and are located in `./build`
+
+// TODO remove test dirs if tests are successful
+
+var (
+	basecoind = "build/basecoind"
+	basecli   = "build/basecli"
+
+	basecoindDir = "./tmp-basecoind-tests"
+	basecliDir   = "./tmp-basecli-tests"
+
+	ACCOUNTS = []string{"alice", "bob", "charlie", "igor"}
+	alice    = ACCOUNTS[0]
+	bob      = ACCOUNTS[1]
+	charlie  = ACCOUNTS[2]
+	igor     = ACCOUNTS[3]
+)
+
+func gopath() string {
+	return filepath.Join(os.Getenv("GOPATH"), "src", "github.com", "cosmos", "cosmos-sdk")
+}
+
+func whereIsBasecoind() string {
+	return filepath.Join(gopath(), basecoind)
+}
+
+func whereIsBasecli() string {
+	return filepath.Join(gopath(), basecli)
+}
+
+func TestInitBaseCoin(t *testing.T) {
+	clean()
+
+	var err error
+
+	password := "some-random-password"
+	usePassword := exec.Command("echo", password)
+
+	initBasecoind := exec.Command(whereIsBasecoind(), "init", "--home", basecoindDir)
+
+	initBasecoind.Stdin, err = usePassword.StdoutPipe()
+	if err != nil {
+		t.Error(err)
+	}
+
+	initBasecoind.Stdout = os.Stdout
+
+	if err := initBasecoind.Start(); err != nil {
+		t.Error(err)
+	}
+	if err := usePassword.Run(); err != nil {
+		t.Error(err)
+	}
+	if err := initBasecoind.Wait(); err != nil {
+		t.Error(err)
+	}
+
+	if err := makeKeys(); err != nil {
+		t.Error(err)
+	}
+}
+
+func makeKeys() error {
+	var err error
+	for _, acc := range ACCOUNTS {
+		pass := exec.Command("echo", "1234567890")
+		makeKeys := exec.Command(whereIsBasecli(), "keys", "add", acc, "--home", basecliDir)
+
+		makeKeys.Stdin, err = pass.StdoutPipe()
+		if err != nil {
+			return err
```

Comment:
Do not return an error, just pass in the `t` and to a good ol' assert.Nil(t, err) - make the code more neat and tighter. also assert uses stack traces :) so doesn't matter that it's in a helper function